sdk/core/System.ClientModel/README.md

```diff
@@ -23,71 +22,290 @@ dotnet add package System.ClientModel
+
+```C# Snippet:ClientModelConfigurationReadme
+MapsClientOptions options = new()
+{
+    NetworkTimeout = TimeSpan.FromSeconds(120),
+};
+
+string? key = Environment.GetEnvironmentVariable("MAPS_API_KEY");
+ApiKeyCredential credential = new(key!);
+MapsClient client = new(new Uri("https://atlas.microsoft.com"), credential, options);
 ```
 
-### Read and write persistable models
+For more information on client configuration, see [Client configuration samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/samples/Configuration.md).
+
+### Customizing HTTP requests
+
+Service clients expose low-level [protocol methods](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/samples/ServiceMethods.md#protocol-methods) that allow callers to customize HTTP requests by passing an optional `RequestOptions` parameter.  `RequestOptions` can be used to modify various aspects of the request sent by the service method, such as adding a request header, or adding a policy to the client pipeline that can modify the request directly before sending it to the service. `RequestOptions` also allows an client user to pass a `CancellationToken` to the method.
 
-As a library author you can implement `IPersistableModel<T>` or `IJsonModel<T>` which will give library users the ability to read and write your models.
+```C# Snippet:RequestOptionsReadme
+// Create RequestOptions instance
+RequestOptions options = new();
 
-Example writing an instance of a model.
+// Set CancellationToken
+options.CancellationToken = cancellationToken;
 
-```C# Snippet:Readme_Write_Simple
-InputModel model = new InputModel();
-BinaryData data = ModelReaderWriter.Write(model);
+// Add a header to the request
+options.AddHeader("CustomHeader", "CustomHeaderValue");
+
+// Call protocol method to pass RequestOptions
+ClientResult output = await client.GetCountryCodeAsync(ipAddress.ToString(), options);
 ```
 
-Example reading a model from json
+For more information on customizing requests, see [Protocol method samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/samples/ServiceMethods.md#protocol-methods).
+
+### Provide request content
+
+In service clients' [protocol methods](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/samples/ServiceMethods.md#protocol-methods), users pass the request content as an input parameter to the method as an instance of `BinaryContent`.  The following example shows how to create a `BinaryContent` instance to pass to a protocol method.
+
+```C# Snippet:ServiceMethodsProtocolMethod
+// Create a BinaryData instance from a JSON string literal.
+BinaryData input = BinaryData.FromString("""
```

Comment:
Today the BinaryContent APIs only support BinaryData and persistable models:
https://apiview.dev/Assemblies/Review/1b123e7a51d44ebe945f0212ee039c65#System.ClientModel.BinaryContent

Is this an ask to move object, string, and JSON writing capabilities into ClientModel, or are you just thinking about calling the existing APIs differently than I'm thinking about it?

Reply:
I was not even thinking that deep, but now that you said that, I think it might be good to add BinaryContent.Create overloads that mimic the binary data ones. I have written some code using SCM, and having to do BinaryContent.Create(BinaryData.FromXxx(...)) always felt as unnecessary ceremony.

Reply:
Got it.  Let's wait on those additions for now unless they are high priority -- @jsquire, with this understanding now, I can address the scenario you flagged in https://github.com/Azure/azure-sdk-for-net/pull/42369#discussion_r1513407132

Reply:
When we have the "infinite stream" input scenario, we will need to move the BinaryContent.Create(Stream) implementation over from Azure.Core.

Reply:
Addressed